core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -95,6 +97,33 @@ export class RubricsEditorComponent {
     return Boolean(this.editableExplanations[difficulty][index]);
   }
 
+  isExplanationLengthValid(difficulty: string, idx: number): boolean {
+    return this.editableExplanations[difficulty][idx].length <=
+    this.maximumCharacterLengthOfExplanation;
+  }
+
+  isMediumLevelExplanationValid(): boolean {
+    // Checking if medium level rubrics have atleast one explantion.
+    this.selectedRubricIndex = 1;
+    if (this.rubrics[this.selectedRubricIndex] === undefined) {
+      return false;
+    }
+    return (this.rubrics[this.selectedRubricIndex].getExplanations().length >=
+    1);
+  }
+
+  isTotalExplanationsLengthReached(): boolean {
+    let totalExplanations: number = 0;
+    for (let idx in this.rubrics) {
+      if (this.rubrics[idx] === undefined) {
+        return false;
+      } else {
```

Comment:
Why are we returning false if any of the rubrics is empty? I'm wondering, what happens if a rubric at index 0 is empty but a rubric at index 2 has >10 explanations? Also, shouldn't this method just be checking the total count of explanations as the method name suggests.

Reply:
Could you also explain when exactly we are expecting an undefined entry in the rubrics array (i.e. what user action causes it to be undefined)?

Reply:
The reason why I have to add `undefined` here is while the time of initialization of all the variables values are assigned to them and by default `undefined` is there so when this function is initialized then it raises an error of `cannot access the property of undefined` because at that particular moment `this.rubrics[]` value is undefined as it's in the phase of initialization, so to pass this particular error I have to perform this action.
Thanks

Reply:
@kevintab95 PTAL

Reply:
Hm, we should not be calling a method if the values that the method operates on is not initialised yet.

Reply:
Okay, I'll look into that and will update you on the same. Thanks for the info.